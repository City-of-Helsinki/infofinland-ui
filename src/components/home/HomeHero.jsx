import Image from 'next/image'

const HomeHero = ({ title, src }) => (
  <div className="relative mb-14 h-homehero md:h-homeheroxl bg-blue-white">
    {src && <Image src={src} layout="fill" alt="" priority objectFit="cover" />}
    <h1 className="absolute bottom-0 mb-16 ml-6 md:max-w-lg text-h1 md:text-h1xl lg:text-h1xl text-white md:ms-12 ifu-hero-home__title text-shadow-home md:min-w-md">
      {title}
    </h1>
  </div>
)

export default HomeHero
