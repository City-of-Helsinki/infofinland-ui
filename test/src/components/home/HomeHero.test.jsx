/**
 * @jest-environment jsdom
 */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react'
import { render } from '@testing-library/react'
import HomeHero from '@/components/home/HomeHero'

describe('HomeHero', () => {
  it('should render the HomeHero component', () => {
    const { asFragment } = render(<HomeHero title="test" image="/foo.png" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
