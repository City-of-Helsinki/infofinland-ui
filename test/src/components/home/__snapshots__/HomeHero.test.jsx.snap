// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeHero should render the HomeHero component 1`] = `
<DocumentFragment>
  <div
    class="relative mb-14 h-homehero md:h-homeheroxl"
    style="background-image: url(/foo.png); background-position: center center; background-size: cover; background-repeat: no-repeat;"
  >
    <h1
      class="absolute bottom-0 mb-16 ml-6 md:max-w-lg text-h1 md:text-h1xl lg:text-h1xl text-white md:ms-12 ifu-hero-home__title text-shadow-home md:min-w-md"
    >
      test
    </h1>
  </div>
</DocumentFragment>
`;
